### PR TITLE
chore: few fixes

### PR DESCRIPTION
**Pull request by tony-go**
- add `EOS` const
- avoid mutate args in functions
- change test names

### CODE REVIEW

**Comment by tony-go** [comment]
@capitnflam ^^

File: src/string_lib.c [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #include <stdio.h>
 
+int EOS = '\0';
```
Comment (capitnflam) [inline src/string_lib.c]: As it is, you just declared a global variable!
You should either add the `static` & `const` keywords or transform this into a macro.

```suggestion
static const int EOS = '\0';
```
or
```suggestion
#define EOS '\0'
```

`static` tells the compiler that this variable is only to be accessed by this compilation unit (file).
`const` has pretty much the same meaning as in JS.

WARNING: `EOS` might mean `End Of Stream` in other contexts.

**Review by capitnflam — APPROVED** [review]
@tony-go LGTM except for `EOS` but this is not a big issue.
I'll have to introduce you to profiling at some point.

File: src/string_lib.c (hunk 1 above)
Comment (tony-go) [reply src/string_lib.c]: Thanks a lot man ^^

**Pull request merged**